hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/DirectoryEntityHandler.java

```diff
@@ -89,7 +91,15 @@ public DUResponse getDuResponse(
     for (long subdirObjectId: subdirs) {
       NSSummary subdirNSSummary =
               getReconNamespaceSummaryManager().getNSSummary(subdirObjectId);
-      String subdirName = subdirNSSummary.getDirName();
+      // for the subdirName we need the filename, not the key name
+      Path subdirPath = Paths.get(subdirNSSummary.getDirName());
+      Path subdirFileName = subdirPath.getFileName();
+      String subdirName;
+      if (subdirFileName != null) {
+        subdirName = subdirFileName.toString();
+      } else {
+        throw new NullPointerException("Subdirectory file name is null.");
+      }
```

Comment:
is this change to remove the trailing slash?  why not: OzoneFSUtils.removeTrailingSlashIfNeeded()

Reply:
My mistake, I updated the comment to be more descriptive of the issue. Eg. the key name is `dir1/dir2`, then we just need to add the subdir to the path, we need `dir2`. Otherwise, we will end up having `dir1` twice while this is an iterative process going over all directories and adding them one by one.

Reply:
So then why test for null and then do an explicit throw?  won't the throw happen even if you don't check for it?

Reply:
I added the null check to get rid of a `findbugs` error but I also didn't want to consume the NPT exception. I can change the exception type to `IOException`.

```
M D NP: Possible null pointer dereference in org.apache.hadoop.ozone.recon.api.handlers.DirectoryEntityHandler.getDuResponse(boolean, boolean) due to return value of called method  Dereferenced at DirectoryEntityHandler.java:[line 102]
```